sdk/notificationhubs/notification-hubs/review/notification-hubs.api.md

```diff
@@ -0,0 +1,177 @@
+## API Report File for "@azure/notification-hubs"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+/// <reference types="node" />
+
+import { CommonClientOptions } from '@azure/core-client';
+import { OperationOptions } from '@azure/core-client';
+import { ServiceClient } from '@azure/core-client';
+
+// @public
+export interface ADMInstallation extends DeviceTokenInstallation {
+    platform: "adm";
+}
+
+// @public
+export interface ADMMessage extends JSONNotificationMessage {
+    platform: "adm";
+}
+
+// @public
+export interface AppleInstallation extends DeviceTokenInstallation {
+    platform: "apple";
+}
+
+// @public
+export interface AppleMessage extends JSONNotificationMessage {
+    platform: "apple";
+}
+
+// @public
+export interface BaiduInstallation extends DeviceTokenInstallation {
+    platform: "baidu";
+}
+
+// @public
+export interface BaiduMessage extends JSONNotificationMessage {
+    platform: "baidu";
+}
+
+// @public
+export interface BrowserInstallation extends InstallationCommon {
+    platform: "browser";
+    pushChannel: BrowserPushChannel;
+}
+
+// @public
+export interface BrowserMessage extends JSONNotificationMessage {
+    platform: "browser";
+}
+
+// @public
+export interface BrowserPushChannel {
+    auth: string;
+    endpoint: string;
+    p256dh: string;
+}
+
+// @public
+export interface DeviceTokenInstallation extends InstallationCommon {
+    pushChannel: string;
+}
+
+// @public
+export interface FirebaseLegacyInstallation extends DeviceTokenInstallation {
+    platform: "gcm";
+}
+
+// @public
+export interface FirebaseLegacyMessage extends JSONNotificationMessage {
+    platform: "gcm";
+}
+
+// @public
+export type Installation = AppleInstallation | ADMInstallation | BaiduInstallation | BrowserInstallation | FirebaseLegacyInstallation | WindowsInstallation;
+
+// @public
+export interface InstallationCommon {
+    readonly expirationTime: string;
+    installationId: string;
+    readonly lastUpdate: string;
+    platform: string;
+    tags?: string[];
+    templates?: Record<string, InstallationTemplate>;
+    userId?: string;
+}
+
+// @public
+export interface InstallationPatch {
+    op: JSONPatchOperation;
+    path: string;
+    value?: string;
+}
+
+// @public
+export interface InstallationTemplate {
+    body: string;
+    expiry?: string;
+    headers: Record<string, string>;
+    tags?: string[];
+}
+
+// @public
+export interface JSONNotificationMessage extends NotificationHubMessageCommon {
+    contentType: "application/json;charset=utf-8";
```

Comment:
do we make consumers pass this in? seems a bit unnecessary if so

Reply:
The only one we cannot infer that is json is WNS.  We could make that required only there

Reply:
I'm also okay if we make what the user passes in something simpler like "json" and we expand it to the full content type... but maybe that's too much magic?